File: Python/InsertionSort.py
def insertionSort(arr):
    i=1
    while i<len(arr):
        if arr[i]<arr[i-1]:
            a = i
            while not a<1:
                if arr[a]>arr[a-1]:
                    break
                tmp = arr[a-1]
                arr[a-1] = arr[a]
                arr[a] = tmp
                a-=1
        i+=1        

arr = [4,23,6,1,5,7,8,20]
print(arr)
print()
insertionSort(arr)
print(arr)

File: Java/InsertionSort.java
import java.util.*;

public class InsertionSort {
    public static void main(String[] args) {
        int[] arr = {4,23,6,1,5,7,8,20};
        System.out.println(Arrays.toString(arr));
        for(int i = 1;i<arr.length;i++){
            if(arr[i]<arr[i-1]){
                int a = i;
                while(!(a<1)){
                    if(arr[a]>arr[a-1]){
                        break;
                    }
                    int temp = arr[a];
                    arr[a]=arr[a-1];
                    arr[a-1]=temp;
                    a--;
                }
            }
        }
        System.out.println(Arrays.toString(arr));
    }
}
